src/main/kotlin/org/rust/cargo/project/configurable/CargoConfigurable.kt

```diff
@@ -6,95 +6,82 @@
+            }
+            row {
+                checkBox(
+                    "Offline mode",
+                    state::useOffline,
+                    comment = "Pass <b>--offline</b> option to Cargo not to perform network requests"
+                )
+            }
+
+            titledRow("External Linter") {
+                subRowIndent = 0
+                row("External tool:") {
+                    comboBox(
+                        EnumComboBoxModel(ExternalLinter::class.java),
+                        state::externalLinter,
+                    ).comment("External tool to use for code analysis")
+                }
+
+                row("Additional arguments:") {
+                    externalLinterArguments(CCFlags.growX)
+                        .comment("Additional arguments to pass to <b>cargo check</b> or <b>cargo clippy</b> command")
+                        .withBinding(
+                            componentGet = { it.text },
+                            componentSet = { component, value -> component.text = value },
+                            modelBinding = state::externalLinterArguments.toBinding()
+                        )
+                }
+                row {
+                    checkBox(
+                        "Run external linter to analyze code on the fly",
+                        state::runExternalLinterOnTheFly,
+                        comment = """
+                            Enable external linter to add code highlighting based on the used linter result.
+                            Can be CPU-consuming
+                        """.trimIndent()
+                    )
+                }
+            }
         }
     }
 
-    override fun isModified(): Boolean =
-        autoUpdateEnabled != settings.autoUpdateEnabled
-            || externalLinter != settings.externalLinter
-            || runExternalLinterOnTheFly != settings.runExternalLinterOnTheFly
-            || compileAllTargets != settings.compileAllTargets
-            || useOffline != settings.useOffline
-            || externalLinterArguments.text != settings.externalLinterArguments
-
     override fun apply() {
-        settings.modify {
-            it.autoUpdateEnabled = autoUpdateEnabled
-            it.externalLinter = externalLinter
-            it.runExternalLinterOnTheFly = runExternalLinterOnTheFly
-            it.externalLinterArguments = externalLinterArguments.text
-            it.compileAllTargets = compileAllTargets
-            it.useOffline = useOffline
-        }
-    }
-
-    override fun reset() {
-        autoUpdateEnabled = settings.autoUpdateEnabled
-        externalLinter = settings.externalLinter
-        runExternalLinterOnTheFly = settings.runExternalLinterOnTheFly
-        externalLinterArguments.text = settings.externalLinterArguments
-        compileAllTargets = settings.compileAllTargets
-        useOffline = settings.useOffline
+        super.apply()
+        project.rustSettings.settingsState = state
```

Comment:
Maybe we can move this line to `CargoConfigurable` somehow? We can introduce an abstract method, something like `validateAll` or `doApply`, and call it in `CargoConfigurable`'s `apply` (which should be `final`).

Reply:
What problem should this change solve?